### ISSUE TITLE
last block ID does not match current chain state

### ISSUE THREAD

**Issue by D3m0nKingx** [DECENTfoundation/DECENT-Network#198]
After adding **miner-id & public/private keys** to the config.ini, on initial blockchain download it syncs up, but after shutting down the daemon and restarting with the new mining settings in the ini file, the following problem occurs: (also I'm not sure I actually have the correct miner-id set? The instruction is somewhat vague on the wiki as where this is obtained from)

https://gist.github.com/D3m0nKingx/2a392325c13d792d442e25d7abd70f58

If I delete the ~/.decent directory and restart it syncs normally again, however the same issue happens upon restart if I add the mining settings again.

**Comment by D3m0nKingx** [comment]
Additionally, this message occurs (on initial run):

`2729982ms th_a       miner.cpp:201                 block_production_loo ] Not producing block because I don't have the private key for DCT8UfuPwryNkLU8gphYbqgiGLHsE7WTSCFffZ7HC8j79fiw8RRtb
`

**Comment by D3m0nKingx** [comment]
Seems the the chain state issue is a result of having the incorrect miner-id in the config.ini, as I just enabled the default miner-id and public/private keys and operation continued normally. 

Thus, where do I obtain my miner-id :question:

**Comment by ejossev** [comment]
There is an error with the undo_db and proposals - in replay mode the undo feature is disabled, but push_proposal force enable it. However it fails on merge. As a workaround, use --resync-blockchain

**Comment by ejossev** [comment]
Regarding IDs' I suggest to generate special key pair just for mining, independent from your ordinary keys. Then run cli method update_miner:
```
suggest_brain_key
{
  "brain_priv_key": "POSITOR EMEER ENHUSK MESIAL GODLY ARAROBA MOUCHE PODLER UPFILL ACRONYC BOBBY BLUR PLAIDY BRECCIA PENHEAD OWNER",
  "wif_priv_key": "5J2Cf17CUKu4QzZZBMnMLzvXfZbh843AudJZd4aBiG6YG24ZhbM",
  "pub_key": "DCT6AL6D8ubuaK5oUN62bLhvxbvjEmXDoCaEvrWBhyZnyEJfdfRM7"
}

update_miner ejossev "ejossev" DCT6AL6D8ubuaK5oUN62bLhvxbvjEmXDoCaEvrWBhyZnyEJfdfRM7 true
```
and add into config.ini the line 
```
private-key = ["DCT6AL6D8ubuaK5oUN62bLhvxbvjEmXDoCaEvrWBhyZnyEJfdfRM7","5J2Cf17CUKu4QzZZBMnMLzvXfZbh843AudJZd4aBiG6YG24ZhbM"]

**Comment by D3m0nKingx** [comment]
Great, that seems to be working now. The parameter that I'm also inquiring to in the config.ini is:

```
# ID of miner controlled by this node (e.g. "1.4.5", quotes are required, may specify multiple times)
miner-id = "1.4.12"
```

What should I be putting for **minder-id =** ?

And ofc I can't do anything with the cli_wallet until we get issue #194 fixed

**Comment by ejossev** [comment]
The miner ID can be get with get_miner <account> command

**Comment by D3m0nKingx** [comment]
Ok, however I can't do any cli commands atm, because cli_wallet isn't working for me right now, due to the other issue still present. But otherwise the daemon is working with the private-key set.

**Comment by Texxer** [comment]
I'm on Windows and I'm stuck at the same step and I have a few additonal questions.

1. How can i check or change my account name if I used suggest_brain_key to make a new one?

If I use the new suggest_brain_key to fill out my config.ini the decentd tell's me there is not enough balance on the DCT address.

2. How do i properly update_miner if I messed up my create_miner?

3. As I understand, I can choose accountname freely. What is the URL used for?

create_miner <accountname> “url-to-proposal” true
